lib/utils.js

```diff
@@ -2,12 +2,32 @@
 
 const normalize = require('path').normalize;
 const IP = require('ip');
+const matcher = require('matcher');
 
+/**
+ * Check whether a domain is in the safe domain white list or not.
+ * Notice: `domain` and `domain_white_list` are case insensitive.
+ * @param {String} domain The inputted domain.
+ * @param {Array<string>} domain_white_list The white list for domain.
+ * @return {Boolean} If the `domain` is in the white list, return true; otherwise false.
+ */
 exports.isSafeDomain = function isSafeDomain(domain, domain_white_list) {
+  // Ignore case sensitive first
+  domain = domain.toLowerCase();
   // add prefix `.`, because all domains in white list are start with `.`
   const hostname = '.' + domain;
+
   return domain_white_list.some(function(rule) {
-    return hostname.endsWith(rule);
+    // Ignore case sensitive
+    rule = rule.toLowerCase();
+    // Check whether we've got '*' as a wild character symbol
+    if (rule.indexOf('*') !== -1) {
```

Comment:
toLowerCase 可以提前处理，不要每次执行的时候做

Reply:
@dead-horse：Do you mean that the user's inputted value should be lower case and there's no need to have a check……or where should I put the codes?

Reply:
OK. Thanks!

Reply:
From your `egg.security` it seems that your `isSafeDomain` only has one parameter and your WhiteList won't be changed since it's assigned with values. So I changed according to your ideas.